main.go

```diff
@@ -176,3 +217,72 @@ func getModulePackageName(modFile *os.File, outputPath string) string {
 	moduleName = strings.TrimSpace(moduleName)
 	return fmt.Sprintf("%v/%v", moduleName, outputPath)
 }
+
+// validate validates the configuration and determines the goPackagePath for the generated code
+func (c *config) validate() error {
```

Comment:
nit, but could rename to `validateAndSetDerivedFields` to convey that it's not just validating

Reply:
just gonna call it `setDerivedFields` because it returns an error so validation is somewhat implied.